logger/src/telemetry.rs

```diff
@@ -82,70 +82,74 @@ impl Telemetry {
+    ) -> (impl Subscriber, mpsc::Receiver<ChannelEvent>) {
         let (sender, receiver) = mpsc::channel(CHANNEL_SIZE);
-        let (sender_future, receiver_future) = mpsc::channel(CHANNEL_SIZE);
-        let telemetry = Self::from_senders(subscriber, sender, sender_future);
-        (telemetry, receiver, receiver_future)
+        let telemetry = Self::from_senders(subscriber, sender);
+        (telemetry, receiver)
     }
 
     /// Create new telemetry layer with specific channel size
     #[allow(clippy::new_ret_no_self)]
     pub fn from_capacity(
         subscriber: S,
         channel_size: usize,
-    ) -> (impl Subscriber, Receiver<Telemetry>, Receiver<Telemetry>) {
+    ) -> (impl Subscriber, mpsc::Receiver<ChannelEvent>) {
         let (sender, receiver) = mpsc::channel(channel_size);
-        let (sender_future, receiver_future) = mpsc::channel(channel_size);
-        let telemetry = Self::from_senders(subscriber, sender, sender_future);
-        (telemetry, receiver, receiver_future)
+        let telemetry = Self::from_senders(subscriber, sender);
+        (telemetry, receiver)
+    }
+
+    fn send_event(&self, channel: Channel, target: &'static str, event: &TracingEvent<'_>) {
+        let _ = self
+            .sender
+            .try_send(ChannelEvent(channel, Event::from_event(target, event)));
     }
 }
 
-impl<S: Subscriber> EventInspectorTrait for TelemetryLayer<S> {
+impl<S: Subscriber> EventInspectorTrait for Layer<S> {
     type Subscriber = S;
 
     fn inner_subscriber(&self) -> &Self::Subscriber {
         &self.subscriber
     }
 
-    fn event(&self, event: &Event<'_>) {
+    fn event(&self, event: &TracingEvent<'_>) {
         let target = event.metadata().target();
         #[allow(clippy::option_if_let_else)] // This is actually more readable.
-        if let Some(telemetry_target) = target.strip_prefix(TELEMETRY_TARGET_PREFIX) {
-            let _result = self
-                .telemetry_sender
-                .try_send(Telemetry::from_event(telemetry_target, event));
-        } else if let Some(future_target) = target.strip_prefix(TELEMETRY_FUTURE_TARGET_PREFIX) {
-            let _result = self
-                .telemetry_future_sender
-                .try_send(Telemetry::from_event(future_target, event));
+        if let Some(target) = target.strip_prefix(TARGET_PREFIX) {
+            self.send_event(Channel::Regular, target, event);
+        } else if let Some(target) = target.strip_prefix(FUTURE_TARGET_PREFIX) {
+            self.send_event(Channel::Future, target, event);
         } else {
             self.subscriber.event(event)
         }
     }
 }
+
+/// A pair of [`Channel`] associated with [`Event`]
+pub struct ChannelEvent(pub Channel, pub Event);
+
+/// Supported telemetry channels
+#[derive(Copy, Clone)]
+pub enum Channel {
+    /// Regular telemetry
+    Regular,
+    /// Future telemetry (?)
+    // FIXME: what is the "future"?
```

Comment:
I think the "future" telemetry is collected from futures instrumented with `iroha_futures::TelemetryFuture`. It logs on every poll and reports how long did it take.

I am not sure if this feature is used by anyone though

Reply:
Updated the docstring